### PR TITLE
Fixed issues in NetNative.

### PR DESCRIPTION
**Pull request by shmao**
Fixes all the places where we use GetDataContractFromGeneratedAssembly.

### CODE REVIEW

File: src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ClassDataContract.cs [hunk 1]
```diff
@@ -282,8 +280,9 @@ internal ConstructorInfo GetNonAttributedTypeConstructor()
             return _helper.GetNonAttributedTypeConstructor();
         }
 
-#if NET_NATIVE
         private XmlFormatClassWriterDelegate _xmlFormatWriterDelegate;
```
Comment (SGuyGe) [inline src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ClassDataContract.cs]: This is not necessary.

File: src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ClassDataContract.cs [hunk 2]
```diff
@@ -319,14 +318,13 @@ internal XmlFormatClassWriterDelegate XmlFormatWriterDelegate
             }
             set
             {
-#if NET_NATIVE
                 _xmlFormatWriterDelegate = value;
-#endif
             }
         }
```
Comment (SGuyGe) [inline src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ClassDataContract.cs]: There's no functional difference here, except removed a #if at the cost of adding a member for NetCore.

File: src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DataContract.cs [hunk 3]
```diff
@@ -100,7 +101,10 @@ internal static DataContract GetDataContractFromGeneratedAssembly(Type type)
                     }
                 }
             }
-            return dataContract;
+            return dataContract;
```
Comment (SGuyGe) [inline src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DataContract.cs]: Nit: Remove trailing spaces.

File: src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ClassDataContract.cs (hunk 2 above)
Comment (shmao) [inline src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ClassDataContract.cs]: I wanted to remove unnecessary #if directives. I've changed this back.

**Comment by shmao** [comment]
I've updated the PR per the feedback.

**Comment by SGuyGe** [comment]
:shipit:

**Pull request merged**